app/workers/principals/delete_job.rb

```diff
@@ -46,9 +47,19 @@ def replace_references(principal)
     Principals::ReplaceReferencesService
       .new
       .call(from: principal, to: DeletedUser.first)
-      .tap do |call|
-      raise ActiveRecord::Rollback if call.failure?
-    end
+      .on_failure { raise ActiveRecord::Rollback }
+  end
+
+  def replace_mentions(principal)
+    # Breaking abstraction here.
+    # Doing the replacement is a very costly operation while at the same time,
+    # groups and placeholder users can't be mentioned.
+    return unless principal.is_a?(User)
```

Comment:
You can @mention groups, and so I'd say their name should also be replaced. I don't think you can mention placeholders right now, though.

Reply:
Done in a0ad5ee